File: backend-dotnet/Models/Review.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ClothingStore.Models
{
    [Table("reviews")]
    public class Review
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }
        
        [Required]
        [Column("user_id")]
        public long UserId { get; set; }
        
        [Required]
        [Column("product_id")]
        public long ProductId { get; set; }
        
        [Required]
        [Column("rating")]
        public int Rating { get; set; }
        
        [Column("comment")]
        [MaxLength(1000)]
        public string? Comment { get; set; }
        
        [Column("review_date")]
        public DateTime ReviewDate { get; set; } = DateTime.Now;
    }
}

File: genaibackend/src/main/java/com/clothingstore/entity/Review.java
package com.clothingstore.entity;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "reviews")
public class Review {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "product_id", nullable = false)
    private Long productId;
    
    @Column(name = "user_id", nullable = false)
    private Long userId;
    
    @Column(nullable = false)
    private Integer rating; // 1-5 stars
    
    private String comment;
    
    @Column(nullable = false)
    private LocalDateTime reviewDate;
    
    // Constructors
    public Review() {
        this.reviewDate = LocalDateTime.now();
    }
    
    public Review(Long productId, Long userId, Integer rating, String comment) {
        this();
        this.productId = productId;
        this.userId = userId;
        this.rating = rating;
        this.comment = comment;
    }
    
    // Getters and Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    
    public Long getProductId() { return productId; }
    public void setProductId(Long productId) { this.productId = productId; }
    
    public Long getUserId() { return userId; }
    public void setUserId(Long userId) { this.userId = userId; }
    
    public Integer getRating() { return rating; }
    public void setRating(Integer rating) { this.rating = rating; }
    
    public String getComment() { return comment; }
    public void setComment(String comment) { this.comment = comment; }
    
    public LocalDateTime getReviewDate() { return reviewDate; }
    public void setReviewDate(LocalDateTime reviewDate) { this.reviewDate = reviewDate; }
}